tests/test_train.py

```diff
@@ -29,33 +29,29 @@ def test_train(cfg, tmpdir):
     cfg_tmp.eval.predict_vids_after_training = True
     cfg_tmp.eval.save_vids_after_training = True
 
-    # change directory to save outputs elsewhere
-    os.chdir(tmpdir)
-
-    # train model
-    train(cfg_tmp)
-
-    # change directory back
-    os.chdir(pwd)
+    # temporarily change working directory to temp output directory
+    with tmp_path:
```

Comment:
:smiley:

Reply:
Best part is - this will reset the working directory even if the training fails. Fixes the issue where when 1 test fails, a bunch of other tests fail due to directories not found.

Reply:
also the tmpdir fixture was deprecated in pytest, and it advised to used tmp_path.

Reply:
turns out this doesn't change working directory. fixed in a subsequent commit.